vars/runATH.groovy

```diff
@@ -198,7 +198,9 @@ private void test(discriminator, commandBase, localSnapshots, localPluginsStashN
     unstashResources(localSnapshots, localPluginsStashName)
     athContainerImage.inside(containerArgs) {
         realtimeJUnit(testResults: 'target/surefire-reports/TEST-*.xml', testDataPublishers: [[$class: 'AttachmentPublisher']]) {
-            def command = './set-java.sh $java_version && eval "$(./vnc.sh)" && export DISPLAY=$BROWSER_DISPLAY && export SHARED_DOCKER_SERVICE=true && export EXERCISEDPLUGINREPORTER=textfile && ' + prepareCommand(commandBase, discriminator, localSnapshots, localPluginsStashName)
+            // Allow call old images without the set-java.sh script. A message is showed if it doesn't exist to alert
+            // that you are using a new pipeline-library with an old ATH source/image (... type: ./set-java.sh: not found)
+            def command = 'if type ./set-java.sh; then ./set-java.sh $java_version; fi; eval "$(./vnc.sh)" && export DISPLAY=$BROWSER_DISPLAY && export SHARED_DOCKER_SERVICE=true && export EXERCISEDPLUGINREPORTER=textfile && ' + prepareCommand(commandBase, discriminator, localSnapshots, localPluginsStashName)
```

Comment:
have you tried `if -x "set-java.sh"`? It means "file exists && executable".

I fill that the command starts to become really hard to follow and understand. Could it be split in different instructions / lines?

Reply:
@alecharp I used **type** because it prints a message indicating that the file is not there (_type: ./set-java.sh: not found_). Finally we have improved a lot the messages written, so I changed the if to your proposal.